infra/modules/network/main.tf

```diff
@@ -14,9 +14,9 @@ module "aws_vpc" {
   azs  = local.availability_zones
   cidr = local.vpc_cidr
 
-  public_subnets             = ["10.0.10.0/24", "10.0.11.0/24", "10.0.12.0/24"]
-  private_subnets            = ["10.0.0.0/24", "10.0.1.0/24", "10.0.2.0/24"]
-  database_subnets           = ["10.0.5.0/24", "10.0.6.0/24", "10.0.7.0/24"]
+  public_subnets             = ["10.${var.second_octet}.10.0/24", "10.${var.second_octet}.11.0/24", "10.${var.second_octet}.12.0/24"]
```

Comment:
Using the `cidrsubnet` function might be another option here if you prefer, instead of string interpolation. https://developer.hashicorp.com/terraform/language/functions/cidrsubnet

Reply:
@jamesbursa I looked into that! But unfortunately I didn't really parse the math TBH.